### ISSUE TITLE
SendAsync is not triggering campaign send?

### ISSUE THREAD

**Issue by robmburke** [brandonseydel/MailChimp.Net#164]
This is my code. Campaign will not send. It is just sitting in MailChimp.
Is there something I'm missing?
According to the checklist on the website, everything is good to go.

    bool IsTestEmail = Convert.ToBoolean(System.Configuration.ConfigurationManager.AppSettings["IsTestEmail"].ToString());

    IMailChimpManager manager = new MailChimpManager("...");
                    
    Campaign c = manager.Campaigns.AddOrUpdateAsync(new Campaign
    {
        Recipients = new Recipient()
        {
            ListId = (IsTestEmail) ? "d12fc0e8f1" : "2ff10b4385"
        },
        Settings = new Setting
        {
            SubjectLine = ((IsTestEmail) ? "[Test] " : string.Empty) + "Vulfpeck Fleet Notification",
            FromName = "Vulfpeck Fleets",
            ReplyTo = "notify@vulfpeck.space",
            AutoFooter = false
        },
        Type = MailChimp.Net.Core.CampaignType.Regular,
    }).Result;

    manager.Content.AddOrUpdateAsync(c.Id, new MailChimp.Net.Core.ContentRequest()
    {
        Html = GenerateEmailHTML(fleet)
    });

    manager.Campaigns.SendAsync(c.Id);

**Comment by ipenekli** [comment]
hi,
if you wanna send test email, you need to use testasync method as follows;
 CampaignTestRequest tr = new CampaignTestRequest();
                                        tr.EmailType = "html";
                                        tr.Emails = emails2test.ToArray();
await mailChimpManager.Campaigns.TestAsync(crmMCCampaignId, tr);

**Comment by robmburke** [comment]
*sigh* So because you saw the word "Test" in my email subject, you immediately send me to something totally unrelated to my problem? So narrow-minded. It's not like having the word "Test" in my email made the program fail because it expected it to be a test. Even if I used the TestAsync, that would not have fixed, or helped me understand, what was wrong with my code.

However, you inadvertently did point me down the correct path. I needed to use `await` in front of the SendAsync to ensure the campaign was completely saved before trying to send it.
